apiserver/logsink/logsink_test.go

```diff
@@ -59,7 +59,7 @@ func (s *logsinkSuite) SetUpTest(c *gc.C) {
 	s.IsolationSuite.SetUpTest(c)
 	s.abort = make(chan struct{})
 	s.written = make(chan params.LogRecord, 1)
-	s.stub.ResetCalls()
+	s.stub = &testing.Stub{}
```

Comment:
Isn't this going to be a -race failure?
Given that we are changing a pointer that would end up accessed by goroutines.

Maybe just move it after the 's.stackMu' ? Though that doesn't cause the individual goroutines to access the mutex before dereferencing the stub.

Reply:
No. Checked under -race locally and it is fine.